dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/internal/utils/IntrospectUserApiUtils.java

```diff
@@ -0,0 +1,120 @@
+        String response = null;
+        try {
+            HttpURLConnection connection = sendUserApiRequest(accessToken, sslSocketFactory);
+            response = readUserApiResponse(connection);
+        } catch (Exception e) {
+            throw new SocialLoginException("INTROSPECT_ERROR_GETTING_USER_INFO", e, new Object[] { e });
+        }
+        return response;
+    }
+
+    HttpURLConnection sendUserApiRequest(@Sensitive String accessToken, SSLSocketFactory sslSocketFactory) throws IOException, SocialLoginException {
+        HttpURLConnection connection = httpUtils.createConnection(HttpUtils.RequestMethod.POST, config.getUserApi(), sslSocketFactory);
+        connection = httpUtils.setHeaders(connection, getUserApiRequestHeaders());
+        connection.setDoOutput(true);
+        
+        OutputStream outputStream = connection.getOutputStream();
+        String postData = "token=" + accessToken;
+        outputStream.write(postData.getBytes());
+        outputStream.close();
+        connection.connect();
+        return connection;
+    }
+
+    @Sensitive
+    Map<String, String> getUserApiRequestHeaders() {
+        Map<String, String> headers = new HashMap<String, String>();
+        String idAndSecretEncoded = Base64Coder.base64Encode(config.getClientId() + ":" + config.getClientSecret());
+
+        headers.put("Authorization", "Basic " + idAndSecretEncoded);
+        headers.put("Accept", "application/json");
+        headers.put("Content-Type", "application/x-www-form-urlencoded");
+        return headers;
+    }
+
+    String readUserApiResponse(HttpURLConnection connection) throws IOException, SocialLoginException {
+        int responseCode = connection.getResponseCode();
+        String response = httpUtils.readConnectionResponse(connection);
+        if (responseCode != HttpServletResponse.SC_OK) {
+            throw new SocialLoginException("INTROSPECT_USER_API_BAD_STATUS", null, new Object[] { responseCode, response });
+        }
+        return modifyExistingResponseToJSON(response);
+    }
+
+    String modifyExistingResponseToJSON(String response) throws SocialLoginException {
+        JsonObject jsonResponse = getJsonResponseIfValid(response);
+        if(jsonResponse.getBoolean("active")) {
+            return jsonResponse.toString();
+        }
+        else {
+            throw new SocialLoginException("INTROSPECT_USER_API_ACTIVE_NOT_TRUE",null, null);
+        }
+
+    }
+
+    private JsonObject getJsonResponseIfValid(String response) throws SocialLoginException {
+        if (response == null || response.isEmpty()) {
+            throw new SocialLoginException("INTROSPECT_USER_API_RESPONSE_NULL_EMPTY", null, null);
+        }
+        try {
+            return Json.createReader(new StringReader(response)).readObject();
+        } catch (JsonParsingException e) {
+            throw new SocialLoginException("INTROSPECT_USER_API_RESPONSE_NOT_JSON", null, new Object[] { response, e });
```

Comment:
I just noticed the code that this was based on had a minor mistake. The second argument to this `SocialLoginException` constructor is an `Exception` that's the cause of this new exception being thrown. That second argument should be the variable `e` instead of `null`.